### PR TITLE
feat: Add vertical bounds calculation

### LINKED ISSUE #228: vertical bounds
Martina requested this feature for setting vertical bounds for pressure levels or for depth similar to `cdo genlevelbounds` if they are present on the dataset.

### PR DESCRIPTION
**Pull request by siligam**
## Summary

This PR adds automatic vertical coordinate bounds calculation for datasets with vertical dimensions (pressure levels, depth, height), similar to CDO's `genlevelbounds` operator. This ensures CMIP compliance for vertical coordinates.

Closes #228

## Features

- **Automatic detection** of common vertical coordinate names:
  - Pressure levels: `plev`, `plev19`, `plev8`, `plev7`, `plev4`, `plev3`, `lev`, `level`, `pressure`
  - Depth: `depth`
  - Height: `height`, `alt`, `altitude`

- **Robust bounds calculation**:
  - Uses midpoint algorithm for interior cells
  - Extrapolates edge cells using adjacent spacing
  - Ensures continuous bounds (no gaps between levels)
  - Works with both regular and irregular vertical grids

- **Pipeline integration**:
  - Can be used as a processing step in pipelines
  - Compatible with existing workflow
  - Preserves existing bounds if already present

### Core Implementation
- **`src/pycmor/std_lib/bounds.py`**: Added `add_vertical_bounds()` function
- **`src/pycmor/std_lib/__init__.py`**: Exported function for pipeline usage

### Tests
- **`tests/unit/test_vertical_bounds.py`**: Comprehensive test suite with 13 test cases covering:
  - Atmospheric pressure levels
  - Ocean depth levels
  - Height coordinates
  - Irregular grids
  - Edge cases (single level, multiple coordinates)
  - Attribute preservation

### Documentation
- **`doc/coordinate_bounds.rst`**: Added comprehensive section on vertical bounds including:
  - Overview and motivation
  - Pipeline usage examples
  - Standalone usage examples
  - Supported coordinate names
  - Ocean depth example

### Examples
- **`examples/vertical_bounds_example.py`**: Working examples demonstrating:
  - Atmospheric pressure levels
  - Ocean depth levels
  - Irregular vertical grids
  - Pipeline integration

### As a Pipeline Step

```python
from pycmor.std_lib import add_vertical_bounds

pipeline = [
    "load_data",
    "get_variable",
    "add_vertical_bounds",  # Add this step
    "convert_units",
    "time_average",
]
```

### Standalone Usage

```python
from pycmor.std_lib.bounds import add_vertical_bounds
import xarray as xr
import numpy as np

# Dataset with pressure levels
ds = xr.Dataset({
    'ta': (['time', 'plev', 'lat', 'lon'], np.random.rand(10, 8, 5, 6)),
}, coords={
    'plev': [100000, 92500, 85000, 70000, 60000, 50000, 40000, 30000],
    'lat': np.linspace(-90, 90, 5),
    'lon': np.linspace(0, 360, 6),
})

# Add vertical bounds
ds_with_bounds = add_vertical_bounds(ds)

# Now ds_with_bounds contains 'plev_bnds'
print(ds_with_bounds['plev_bnds'])
```

## Testing

All tests pass successfully:

```bash
$ pytest tests/unit/test_vertical_bounds.py -v
====================== 13 passed, 3 warnings in 0.84s ======================
```

Test coverage includes:
- ✅ Standard CMIP6 pressure levels
- ✅ Ocean depth levels
- ✅ Height coordinates
- ✅ Irregular vertical grids
- ✅ Single level edge case
- ✅ Multiple vertical coordinates
- ✅ Existing bounds preservation
- ✅ Attribute preservation
- ✅ Bounds continuity verification

## Code Quality

- ✅ All code formatted with `black`
- ✅ Imports sorted with `isort`
- ✅ Passes `flake8` linting
- ✅ Comprehensive docstrings
- ✅ Type hints included

## Comparison to CDO

This implementation provides similar functionality to CDO's `genlevelbounds` operator:

```bash
# CDO equivalent
cdo genlevelbounds input.nc output.nc

# Pymorize equivalent
ds_with_bounds = add_vertical_bounds(ds)
```

## Benefits

1. **CMIP Compliance**: Ensures vertical coordinates have required bounds
2. **Automatic**: No manual bounds calculation needed
3. **Flexible**: Works with various vertical coordinate types
4. **Robust**: Handles irregular grids and edge cases
5. **Well-tested**: Comprehensive test coverage
6. **Documented**: Full documentation and examples provided

## Checklist

- [x] Code implemented and tested
- [x] All tests passing (13/13)
- [x] Documentation updated
- [x] Examples provided
- [x] Code formatted and linted
- [x] Addresses issue #228

### CODE REVIEW

File: src/pycmor/std_lib/bounds.py [hunk 1]
```diff
@@ -219,6 +219,105 @@ def add_bounds_from_coords(
...
+            "depth",
+            "plev19",
+            "plev8",
+            "plev7",
+            "plev4",
+            "plev3",
+            "height",
+            "alt",
+            "altitude",
+        ]
```
[the remark this answers is not in the document]
Comment (siligam) [reply src/pycmor/std_lib/bounds.py]: i suspect this list undergoes refactoring later on. I propose that we fix it after couple of iterations when we or Martina uses on real data. Anyways my first impression after scanning for dimension names named in CMIP7, I am certain that we might consider improving this code.
